### DISCUSSION TITLE
How to disable editing dynamically imported excel cells?

### DISCUSSION THREAD

**Discussion by TadasMil** [dream-num/Luckysheet#1350] [General]
I have excel file that I pass into `LuckyExcel.transformExcelToLucky`. The file gets rendered correctly and I see data inside the cells. 

I want to make my spreadsheet read-only, thus I pass in the `authority` configuration the following way:
```
const mapConfig = (data) => {
  const sheetConfig = data.sheets.map((sheet) => {
    return {
      ...sheet,
      config: {
        ...sheet.config,
        authority: {
          selectLockedCells: 0,
          formatCells: 0,
          formatColumns: 0,
          formatRows: 0,
          insertColumns: 0,
          insertRows: 0,
          insertHyperlinks: 0,
          deleteColumns: 0,
          deleteRows: 0,
          sort: 0,
          filter: 0,
          usePivotTablereports: 0,
          editObjects: 0,
          editScenarios: 0,
          sheet: 1,
        },
      },
    };
  });


  return sheetConfig;
};
```

The cells with no data are read-only, but the cells that have data in it (from the excel file) are able to be modified. 

Is this a possible bug, or my `authority` configuration is not setup correctly? Or are there any other config options, that should disable modifying cells that include data?

**Comment by digiupdatehub** [comment]
Best Video Editing Software for Free on pc
https://www.digiupdatehub.tech/2023/06/best-video-editing-software-for-free-on.html
